In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings("ignore")

In [2]:
df = pd.read_csv("advertising.csv")
df.head()

Unnamed: 0     TV  radio  newspaper  sales
0           1  230.1   37.8       69.2   22.1
1           2   44.5   39.3       45.1   10.4
2           3   17.2   45.9       69.3    9.3
3           4  151.5   41.3       58.5   18.5
4           5  180.8   10.8       58.4   12.9

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  200 non-null    int64  
 1   TV          200 non-null    float64
 2   radio       200 non-null    float64
 3   newspaper   200 non-null    float64
 4   sales       200 non-null    float64
dtypes: float64(4), int64(1)
memory usage: 7.9 KB


In [11]:
df.describe()

Unnamed: 0          TV       radio   newspaper       sales
count  200.000000  200.000000  200.000000  200.000000  200.000000
mean   100.500000  147.042500   23.264000   30.554000   14.022500
std     57.879185   85.854236   14.846809   21.778621    5.217457
min      1.000000    0.700000    0.000000    0.300000    1.600000
25%     50.750000   74.375000    9.975000   12.750000   10.375000
50%    100.500000  149.750000   22.900000   25.750000   12.900000
75%    150.250000  218.825000   36.525000   45.100000   17.400000
max    200.000000  296.400000   49.600000  114.000000   27.000000

In [4]:
x = df.drop(["Unnamed: 0", "sales"], axis=1)
y = df["sales"]

In [5]:
x

TV  radio  newspaper
0    230.1   37.8       69.2
1     44.5   39.3       45.1
2     17.2   45.9       69.3
3    151.5   41.3       58.5
4    180.8   10.8       58.4
..     ...    ...        ...
195   38.2    3.7       13.8
196   94.2    4.9        8.1
197  177.0    9.3        6.4
198  283.6   42.0       66.2
199  232.1    8.6        8.7

[200 rows x 3 columns]

In [6]:
y

0      22.1
1      10.4
2       9.3
3      18.5
4      12.9
       ... 
195     7.6
196     9.7
197    12.8
198    25.5
199    13.4
Name: sales, Length: 200, dtype: float64

In [7]:
from sklearn.model_selection import train_test_split
xtrain, xtest, ytrain, ytest = train_test_split(x,y, test_size=0.25, random_state=1)

from sklearn.linear_model import LinearRegression
linreg = LinearRegression()
linreg.fit(xtrain, ytrain)
ypred = linreg.predict(xtest)

from sklearn.metrics import r2_score, mean_squared_error, mean_absolute_error
print(f"MAE : {mean_absolute_error(ytest, ypred)}")
print(f"MSE : {mean_squared_error(ytest, ypred)}")
print(f"RMSE: {np.sqrt(mean_squared_error(ytest, ypred))}")
print(f"Accuracy : {r2_score(ytest, ypred)}")

MAE : 1.0668917082595215
MSE : 1.9730456202283397
RMSE: 1.404651423032896
Accuracy : 0.9156213613792231


In [8]:
linreg.coef_

array([0.04656457, 0.17915812, 0.00345046])

In [9]:
x = df.drop(["Unnamed: 0", "sales"], axis=1)
y = df["sales"]

In [10]:
coef_df = pd.DataFrame(linreg.coef_, x.columns, columns=["Coefficient"])
coef_df

Coefficient
TV            0.046565
radio         0.179158
newspaper     0.003450

- For every unit growth TV Advertising, we estimate that Sales will grow by 0.046565
- For every unit growth Radio Advertising, we estimate that Sales will grow by 0.179158
- For every unit growth newspaper Advertising, we estimate that Sales will grow by 0.003450